src/analytic_engine/src/compaction/runner/remote_runner.rs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use async_trait::async_trait;
+use generic_error::BoxError;
+use snafu::ResultExt;
+
+use super::{local_runner::LocalCompactionRunner, node_picker::RemoteCompactionNodePickerRef};
+use crate::{
+    compaction::runner::{
+        remote_client::{build_compaction_client, CompactionClientConfig, CompactionClientRef},
+        CompactionRunner, CompactionRunnerResult, CompactionRunnerTask,
+    },
+    instance::flush_compaction::{
+        BuildCompactionClientFailed, ConvertCompactionTaskResponse, GetCompactionClientFailed,
+        PickCompactionNodeFailed, Result,
+    },
+};
+
+pub struct RemoteCompactionRunner {
+    pub node_picker: RemoteCompactionNodePickerRef,
+    /// Responsible for executing compaction task locally if fail to remote
+    /// compact, used for better fault tolerance.
+    pub local_compaction_runner: LocalCompactionRunner,
+}
+
+impl RemoteCompactionRunner {
+    async fn get_compaction_client(&self) -> Result<CompactionClientRef> {
+        let mut config = CompactionClientConfig::default();
+        let node_addr = self
```

Comment:
A `http://` prefix is necessary for the node_addr.
We can refer to the `make_formatted_endpoint` function in remote engine client.